dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt

```diff
@@ -158,6 +160,27 @@ internal object CoreFeature {
         )
     }
 
+    fun drainAndShutdownExecutors() {
+        val tasks = ArrayList<Runnable>()
+        (persistenceExecutorService as? ThreadPoolExecutor)
+            ?.queue
+            ?.drainTo(tasks)
+        // we make sure we upload the currently locked files
+        uploadExecutorService
+            .queue
+            .drainTo(tasks)
+        // we need to make sure we drain the runnables in both executors first
+        // then we shut them down by using the await termination method to make sure we block
+        // the thread until the active task is finished.
+        persistenceExecutorService.shutdown()
```

Comment:
how we are going to re-create schedulers after?  it seems the main problem is `upload` executor, which does the recurring scheduling. what if instead shutting it down we introduce some lock/condition, which can be checked if it should re-schedule task, which we may activate during flushing operation. like wrapper around executor, where we can put it in the `paused` state. wdyt?

Reply:
The main reason for this method is to flush the sdk and stop the executor. I added all the required docs in the `Datadog#flushAndShutdownExecutors` method explaining that this will leave the SDK in an inconsistent state without being able to use it. You will not be able to re - use this SDK session, you will have to re - initialize it, my intention was previously to not shutdown the executors but unfortunately we need this to make sure we do not send duplicated data.